libs/libc/semaphore/sem_init.c

```diff
@@ -79,7 +79,7 @@ int nxsem_init(FAR sem_t *sem, int pshared, unsigned int value)
   sem->flags = 0;
 
 #ifdef CONFIG_PRIORITY_INHERITANCE
-#  if CONFIG_SEM_PREALLOCHOLDERS > 0
+#  if defined(CONFIG_SEM_PREALLOCHOLDERS) && CONFIG_SEM_PREALLOCHOLDERS > 0
```

Comment:
move below define to `include/semaphore.h`
https://github.com/apache/nuttx/blob/master/sched/semaphore/sem_holder.c#L45-L47